Sources/Apodini/Semantic Model Builder/EndpointsTree.swift

```diff
@@ -18,14 +18,18 @@ struct EndpointRelationship { // ... to be replaced by a proper Relationship mod
 }
 
 /// Models a single Endpoint which is identified by its PathComponents and its operation
-struct Endpoint {
+struct Endpoint: Hashable, Equatable {
     /// This is a reference to the node where the endpoint is located
     // swiftlint:disable:next implicitly_unwrapped_optional
     fileprivate var treeNode: EndpointsTreeNode!
     
     /// Description of the associated component, currently included for debug purposes
     let description: String
     
+    /// An identifier which uniquely identifies this endpoint (via its handler)
+    /// across multiple compilations and executions of the web service.
+    let identifier: AnyHandlerIdentifier
```

Comment:
I really like the current approach of uniquely identifying an endpoint over different compilations and executions of the web service. I have also started to make some thoughts in this regard, because for my thesis I will be detecting api changes between different versions of the web service. However, something that I can observe from the PR, and perhaps might be interesting also for you. Currently, this property does uniquely identify the endpoint, **IFF** the order of registering the components, remains the same over multiple compilations and executions of the web service.

Reply:
Yeah you're right in that regard. My primary use case here is being able to "match" the same instance of a handler across multiple simultaneously running instances.
For example, if your web service (consisting of 3 handlers A, B, C) is deployed to two instances, and you want to remotely invoke a specific handler in the other instance, we need a way to find the correct handler in that other process.
This will also be used to determine which instances of your deployed system should implement which handlers.

As currently implemented, the identifiers are only "stable" as long as the web service's structure remains unchanged.

Another thing is that the identifier's `rawValue` is in itself pretty meaningless: it's either some user-provided string, for `IdentifiableHandler`s, or an IndexPath-like string for all other handlers. (eg: `0:1:0:3`, which would indicate "the third child of the first child of the second child of the root node".)
The obvious issue here is that these IndexPath-like identifiers can change even if the exported interface doesn't change (eg: you wrap a component in a Group w/out any path parameters, or you add additional children to a group).

Reply:
Thanks for the detailed explanation. I can see that the current approach satisfies your use case. Just wanted to point that out, if that would have been relevant for your work as well. Good job overall  👏

Reply:
My assumption here is that, if a web service is deployed as a system of multiple instances, all of these must always be deployed together. That way we eliminate the issue of identifier mismatches between multiple deployments.

I think it should be possible to track `IdentifiableHandler`s across compilations when the structure of the web service changed. As long as the handler's custom identifier remains that same, it should be sufficient to compare the IndexPath to see if it was moved w/in the DSL, and then based on the path components of the generated Endpoint determine if the move somehow changed the exported interface.